core/src/main/java/org/keycloak/sdjwt/SdJwtVerificationContext.java

```diff
@@ -160,16 +164,23 @@ public void verifyPresentation(
      *
      * @throws VerificationException if verification failed
      */
-    private void validateIssuerSignedJwt(SignatureVerifierContext verifier) throws VerificationException {
+    private void validateIssuerSignedJwt(
+            List<SignatureVerifierContext> verifiers
+    ) throws VerificationException {
         // Check that the _sd_alg claim value is understood and the hash algorithm is deemed secure
         issuerSignedJwt.verifySdHashAlgorithm();
 
         // Validate the signature over the Issuer-signed JWT
-        try {
-            issuerSignedJwt.verifySignature(verifier);
-        } catch (VerificationException e) {
-            throw new VerificationException("Invalid Issuer-Signed JWT", e);
+        for (SignatureVerifierContext verifier : verifiers) {
+            try {
+                issuerSignedJwt.verifySignature(verifier);
+                return;
+            } catch (VerificationException ignored) {
+            }
```

Comment:
I think  a debug log could be helpful in the catch.

Reply:
Thanks. Added log messages.